src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -225,13 +228,48 @@ public static boolean calledByAnIncompleteSyntheticClass(MethodCallExpr method)
    * @param method the method call to be analyzed
    * @return the name of the synthetic class of that method
    */
-  public static String getSyntheticClass(MethodCallExpr method) {
+  public static @ClassGetSimpleName String getSyntheticClass(MethodCallExpr method) {
     // if calledByAnIncompleteSyntheticClass returns true for this method call, we know that it has
     // a caller.
-    String fullNameOfTheClass = method.getScope().get().calculateResolvedType().describe();
-    String shortNameOfTheClass =
-        fullNameOfTheClass.substring(fullNameOfTheClass.lastIndexOf('.') + 1);
-    return shortNameOfTheClass;
+    ResolvedType callerExpression = method.getScope().get().calculateResolvedType();
+    if (callerExpression instanceof ResolvedReferenceType) {
+      ResolvedReferenceType referCaller = (ResolvedReferenceType) callerExpression;
+      @FullyQualifiedName String callerName = referCaller.getQualifiedName();
+      @ClassGetSimpleName String callerSimple = fullyQualifiedToSimple(callerName);
+      return callerSimple;
+    } else {
+      throw new RuntimeException("Unexpected expression: " + callerExpression);
+    }
+  }
+
+  /**
+   * This method converts a @FullyQualifiedName classname to a @ClassGetSimpleName classname. Note
+   * that there is warning suppression here. It is safe to claim that if we split
+   * a @FullyQualifiedName name by dot ([.]), then the last part is the @ClassGetSimpleName part.
+   *
+   * @param fullyQualifiedName a @FullyQualifiedName classname
+   * @return the @ClassGetSimpleName version of that class
+   */
+  public static @ClassGetSimpleName String fullyQualifiedToSimple(
+      @FullyQualifiedName String fullyQualifiedName) {
+    @SuppressWarnings("signature")
+    @ClassGetSimpleName String simpleName = fullyQualifiedName.substring(fullyQualifiedName.lastIndexOf(".") + 1);
```

Comment:
If you move the `@SuppressWarnings` to the method's declaration, you can remove the `@ClassGetSimpleName` annotation on this line.